File: mobidic_spring/src/main/java/com/kimtaeyang/mobidic/dto/VocabDto.java
package com.kimtaeyang.mobidic.dto;

import com.kimtaeyang.mobidic.entity.Vocab;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class VocabDto {
    private UUID id;
    private UUID memberId;
    private String title;
    private String description;
    private Timestamp createdAt;

    public static VocabDto fromEntity (Vocab vocab) {
        return VocabDto.builder()
                .title(vocab.getTitle())
                .memberId(vocab.getMember().getId())
                .id(vocab.getId())
                .description(vocab.getDescription())
                .createdAt(vocab.getCreatedAt())
                .build();
    }
}

File: mobidic_flutter/lib/dto/vocab_dto.dart
class VocabDto {
  final String id;
  final String memberId;
  final String title;
  final String description;
  final DateTime? createdAt;

  VocabDto({
    required this.id,
    required this.memberId,
    required this.title,
    required this.description,
    required this.createdAt,
  });

  factory VocabDto.fromJson(Map<String, dynamic> json) => VocabDto(
    id: json['id'],
    memberId: json['memberId'],
    title: json['title'],
    description: json['description'],
    createdAt:
        json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
  );
}
